[PT2][Optimus] Fix a corner case in merge splits

mengluy0125 opened the pull request:
Summary:
We observed another corner case where not all split items are used, see the screenshot

{F1960315622}

We thus skip such cases by checking the getitem indices.

Test Plan:
# local reproduce
```
buck2 run mode/opt //scripts/jackiexu0313/pt2:local_model_with_pt2 -- --test_mode batch-split  --flow_id 663157369 2>&1 | tee ~/cmf.txt
```
P1679677122

# E2E

before fix
f663157369

after fix

Differential Revision: D65990213




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

zeshengzong commented on torch/_inductor/fx_passes/split_cat.py:
```diff
@@ -631,6 +631,10 @@ def merge_splits(
                 }
                 # It is not necessary all getitems from the split node are used.
                 # We use the num of users to check the getitems to be merged.
```
Seems this is the comment for line 638 :D

mengluy0125 committed:
[PT2][Optimus] Fix a corner case in merge splits (#140788)

Summary:

We observed another corner case where not all split items are used, see the screenshot

{F1960315622}

We thus skip such cases by checking the getitem indices.

Test Plan:
# local reproduce
```
buck2 run mode/opt //scripts/jackiexu0313/pt2:local_model_with_pt2 -- --test_mode batch-split  --flow_id 663157369 2>&1 | tee ~/cmf.txt
```
P1679677122

# E2E

before fix
f663157369

after fix

Differential Revision: D65990213
